### ISSUE TITLE
API version in the data (batch/v1beta1) does not match the expected API version (batch/v1)

### ISSUE THREAD

**Issue by tmaior** [keylimetoolbox/cron-kubernetes-ruby#11]
Hey there,
I'm trying to deploy cronjobs on Kubernetes with the gem for our rails app, but when updating the cronjobs we receive an error about batch API version. The first time we run the bundle exec command(when there are no cronjobs running) this error doesn't happen.

`bundle exec cron_kubernetes --configuration config/initializers/cron_kubernetes.rb --schedule config/schedule.rb`

```
deploy@backend-6f5c84f4fd-gt6pd:/srv$ bundle exec cron_kubernetes --configuration config/initializers/cron_kubernetes.rb --schedule config/schedule.rb
bundler: failed to load command: cron_kubernetes (/srv/vendor/bundle/ruby/2.6.0/bin/cron_kubernetes)
Kubeclient::HttpError: the API version in the data (batch/v1beta1) does not match the expected API version (batch/v1)
  /srv/vendor/bundle/ruby/2.6.0/gems/kubeclient-4.10.1/lib/kubeclient/common.rb:130:in `rescue in handle_exception'
  /srv/vendor/bundle/ruby/2.6.0/gems/kubeclient-4.10.1/lib/kubeclient/common.rb:120:in `handle_exception'
  /srv/vendor/bundle/ruby/2.6.0/gems/kubeclient-4.10.1/lib/kubeclient/common.rb:416:in `update_entity'
  /srv/vendor/bundle/ruby/2.6.0/gems/kubeclient-4.10.1/lib/kubeclient/common.rb:257:in `block (2 levels) in define_entity_methods'
  /srv/vendor/bundle/ruby/2.6.0/gems/cron-kubernetes-1.1.0/lib/cron_kubernetes/cron_tab.rb:59:in `update_cron_job'
  /srv/vendor/bundle/ruby/2.6.0/gems/cron-kubernetes-1.1.0/lib/cron_kubernetes/cron_tab.rb:15:in `block in update'
  /srv/vendor/bundle/ruby/2.6.0/gems/cron-kubernetes-1.1.0/lib/cron_kubernetes/cron_tab.rb:15:in `each'
  /srv/vendor/bundle/ruby/2.6.0/gems/cron-kubernetes-1.1.0/lib/cron_kubernetes/cron_tab.rb:15:in `update'
  /srv/vendor/bundle/ruby/2.6.0/gems/cron-kubernetes-1.1.0/bin/cron_kubernetes:29:in `<top (required)>'
  /srv/vendor/bundle/ruby/2.6.0/bin/cron_kubernetes:23:in `load'
  /srv/vendor/bundle/ruby/2.6.0/bin/cron_kubernetes:23:in `<top (required)>'
```
cron_kubernetes.rb:
```
CronKubernetes.configuration do |config|
    # Required
    config.identifier   = "my-crons"
    config.manifest     = YAML.load_file(File.join("/srv/deploy", "kubernetes-job.yml"))
  
    # Optional
    config.output       = nil
    # config.job_template = %w[/bin/bash -l -c :job]
    auth_options = {
        bearer_token_file: '/var/run/secrets/kubernetes.io/serviceaccount/token'
    }
    ssl_options = {}
    if File.exist?("/var/run/secrets/kubernetes.io/serviceaccount/ca.crt")
        ssl_options[:ca_file] = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
    end
    namespace = File.read('/var/run/secrets/kubernetes.io/serviceaccount/namespace')
    config.kubeclient = Kubeclient::Client.new(
        'https://kubernetes.default.svc/apis/batch',
        'v1',
        auth_options: auth_options,
        ssl_options:  ssl_options
      )
  end
```
kubernetes-job.yml
```
apiVersion: batch/v1
kind: Job
metadata:
  name: my-app-job
  namespace: my-app-crons
spec:
  template:
    metadata:
      name: my-app-job
    spec:
      containers:
        - name: rails
          image: my-app-image
          envFrom:
            - secretRef:
                name: my-secret
            - configMapRef:
                name: my-configmap
      restartPolicy: OnFailure
      revisionHistoryLimit: 3

```

Versions:
```
cron-kubernetes: 1.1.0
kubeclient: 4.10.1
ruby: 2.6.5
kubernetes: 1.28
```

**Comment by jeremywadsack** [comment]
Hi,

We fixed the API reference in commit 344ffdea52f231a6ec1cc5e7b947f15478324c1a, PR #9, which is included in `cron-kubernetes` v2.0.0. It looks like you are running v1.1.0. If you upgrade I believe that should resolve this for you. You will need to upgrade to Ruby 2.7 to support that update.

**Comment by tmaior** [comment]
> Hi,
> 
> We fixed the API reference in commit [344ffde](https://github.com/keylimetoolbox/cron-kubernetes-ruby/commit/344ffdea52f231a6ec1cc5e7b947f15478324c1a), PR #9, which is included in `cron-kubernetes` v2.0.0. It looks like you are running v1.1.0. If you upgrade I believe that should resolve this for you. You will need to upgrade to Ruby 2.7 to support that update.


Looks like the version with `v1beta1` was the issue as you said. Thank you!
